source/AI.cpp

```diff
@@ -3624,7 +3624,7 @@ bool AI::Has(const Ship &ship, const Government *government, int type) const
 
 
 
-void AI::UpdateStrengths(map<const Government *, int64_t> &strength, const System *playerSystem)
+void AI::UpdateStrengths(const PlayerInfo &player, map<const Government *, int64_t> &strength, const System *playerSystem)
```

Comment:
Functions that get the full playerInfo class as parameter don't seem to need the playerSystem as a separate parameter (since they can get the playerSystem from playerInfo).

Reply:
I've changed this function and Person::Frequency to pass just the player instead of the player and the player's system. All other functions that are changed though need to remain the same, either because they get a planet or system that isn't the player's current planet or system, or there are multiple functions of the same name that are only differentiated by being given a planet or system.

Reply:
> because they get a planet or system that isn't the player's current planet or system

This is a good reason I agree with.

> or there are multiple functions of the same name that are only differentiated by being given a planet or system.

This I disagree with; function overloading is not a goal in itself and if we need to give duplicate data in parameters to make it work, then we need to fix this in a different way (for example by not overloading using a single name, but by using different function names, or by adding a new function in `LocationFilter` that has as signature `bool Matches(const PlayerInfo &player, const System *origin = nullptr) const;)` besides the existing Matching functions.

I also wonder now if we really always need to pass PlayerInfo all the way to LocationFilter. some examples:
- If we want to know if some government can enforce fines in some system, then can we ignore the "visited system" and "visited planet" checks, since the check only happens when the player is already present in the relevant system or on the relevant planet?
- If we want to know if some government can scan the player, then isn't the player also already in the system when the check is relevant?